src/nxt_http_route.c

```diff
@@ -2071,8 +2079,8 @@ nxt_http_route_test_rule(nxt_http_request_t *r, nxt_http_route_rule_t *rule,
 
 
 static nxt_int_t
-nxt_http_route_pattern(nxt_http_request_t *r, nxt_http_route_pattern_t *pattern,
-    u_char *start, size_t length)
+nxt_http_route_pattern(nxt_task_t *task, nxt_http_request_t *r,
+    nxt_http_route_pattern_t *pattern, u_char *start, size_t length)
 {
     u_char                          *p, *end, *test;
     size_t                          test_length;
```

Comment:
To make it easier for you to find in the patches, the actual logging changes are here.  Please review this function, and maybe suggest some text improvements.

Reply:
Well, github isn't showing the entire function nicely.  I'll paste it here:

```c
$ grepc -tfd nxt_http_route_pattern
./src/nxt_http_route.c:2081:
static nxt_int_t
nxt_http_route_pattern(nxt_task_t *task, nxt_http_request_t *r,
    nxt_http_route_pattern_t *pattern, u_char *start, size_t length)
{
    u_char                          *p, *end, *test;
    size_t                          test_length;
    uint32_t                        i;
    nxt_array_t                     *pattern_slices;
    nxt_http_route_pattern_slice_t  *pattern_slice;

#if (NXT_HAVE_REGEX)
    if (pattern->regex) {
        if (r->regex_match == NULL) {
            r->regex_match = nxt_regex_match_create(r->mem_pool, 0);
            if (nxt_slow_path(r->regex_match == NULL)) {
                return NXT_ERROR;
            }
        }
        nxt_debug(task, "match try regex: \"%V\"", &pattern->u.regex->pattern);

        return nxt_regex_match(pattern->u.regex, start, length, r->regex_match);
    }
#endif

    if (length < pattern->min_length) {
        nxt_debug(task, "match: string is short");
        return 0;
    }

    nxt_assert(pattern->u.pattern_slices != NULL);

    pattern_slices = pattern->u.pattern_slices;
    pattern_slice = pattern_slices->elts;
    end = start + length;

    for (i = 0; i < pattern_slices->nelts; i++, pattern_slice++) {
        test = pattern_slice->start;
        test_length = pattern_slice->length;

        switch (pattern_slice->type) {
        case NXT_HTTP_ROUTE_PATTERN_EXACT:
            nxt_debug(task, "match try exact: \"%*s\"", test_length, test);
            return ((length == pattern->min_length) &&
                    nxt_http_route_memcmp(start, test, test_length,
                                          pattern->case_sensitive));

        case NXT_HTTP_ROUTE_PATTERN_BEGIN:
            nxt_debug(task, "match try begin: \"%*s\"", test_length, test);
            if (nxt_http_route_memcmp(start, test, test_length,
                                      pattern->case_sensitive))
            {
                start += test_length;
                break;
            }

            return 0;

        case NXT_HTTP_ROUTE_PATTERN_END:
            nxt_debug(task, "match try end: \"%*s\"", test_length, test);
            p = end - test_length;

            if (nxt_http_route_memcmp(p, test, test_length,
                                      pattern->case_sensitive))
            {
                end = p;
                break;
            }

            return 0;

        case NXT_HTTP_ROUTE_PATTERN_SUBSTRING:
            nxt_debug(task, "match try substring: \"%*s\"", test_length, test);
            if (pattern->case_sensitive) {
                p = nxt_memstrn(start, end, (char *) test, test_length);

            } else {
                p = nxt_memcasestrn(start, end, (char *) test, test_length);
            }

            if (p == NULL) {
                return 0;
            }

            start = p + test_length;
        }
    }

    return 1;
}
```